crates/red_knot_project/tests/check.rs

```diff
@@ -279,23 +279,4 @@ impl SourceOrderVisitor<'_> for PullTypesVisitor<'_> {
 
 /// Whether or not the .py/.pyi version of this file is expected to fail
 #[rustfmt::skip]
-const KNOWN_FAILURES: &[(&str, bool, bool)] = &[
-    // related to circular references in nested functions
-    ("crates/ruff_linter/resources/test/fixtures/flake8_return/RET503.py", false, true),
-    // related to circular references in class definitions
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_26.py", true, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_27.py", true, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F811_19.py", true, false),
-    ("crates/ruff_linter/resources/test/fixtures/pyupgrade/UP039.py", true, false),
-    // related to circular references in type aliases (salsa cycle panic):
-    ("crates/ruff_python_parser/resources/inline/err/type_alias_invalid_value_expr.py", true, true),
-    ("crates/ruff_linter/resources/test/fixtures/flake8_type_checking/TC008.py", true, true),
-    // related to circular references in f-string annotations (invalid syntax)
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_15.py", true, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F821_14.py", false, true),
-    // related to circular references in stub type annotations (salsa cycle panic):
-    ("crates/ruff_linter/resources/test/fixtures/pycodestyle/E501_4.py", false, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F401_0.py", false, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F401_12.py", false, true),
-    ("crates/ruff_linter/resources/test/fixtures/pyflakes/F401_14.py", false, true),
-];
+const KNOWN_FAILURES: &[(&str, bool, bool)] = &[];
```

Comment:
Should we delete this functionality now that all known failures are fixed? Do we forsee situations where it is okay for a PR to introduce a new panic case over fixing the panic?

Reply:
It's a good question. I thought we could keep it for now. I _think_ any new cycle cases we run into should be easy to fix by adding cycle handling for a new query, but it's possible we could run into such a case where it's difficult to get the cycle handling correct and it makes more sense to split it into a separate PR, in which scenario we might still use this? I don't feel strongly.

Reply:
That makes sense. Although that also likely means that we would start regressing on projects that Red Knot supported before. So maybe the correct order would be to fix the cycle first, and then land the behavior change?

I don't feel strongly either and am happy to go either way. It just feels nice if regressing panics is *not easy*

Reply:
> we would start regressing on projects that Red Knot supported before

Maybe; it depends whether the PR is actually regressing on existing files (in this case I would be more hesitant to allow using `KNOWN_FAILURES`), or if it is e.g. an unrelated Ruff linter PR adding a new test file, which reveals an existing-but-not-previously-tested cycle. This latter case is I think the strongest reason to keep `KNOWN_FAILURES` around for now.